pkg/controller/installation/products/solutionexplorer/reconciler.go

```diff
@@ -87,20 +87,48 @@ func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation,
 	if err != nil {
 		return v1alpha1.PhaseFailed, err
 	}
-	phase, err = r.ReconcileOauthClient(ctx, inst, &v1.OAuthClient{
+
+	if r.Config.GetHost() != route {
+		r.Config.SetHost(route)
+		r.ConfigManager.WriteConfig(r.Config)
+	}
+
+	phase, err = r.ReconcileOauthClient(ctx, inst, &oauthv1.OAuthClient{
 		RedirectURIs: []string{route},
 		Secret:       "test",
-		GrantMethod:  v1.GrantHandlerAuto,
-		ObjectMeta: v12.ObjectMeta{
+		GrantMethod:  oauthv1.GrantHandlerAuto,
+		ObjectMeta: metav1.ObjectMeta{
 			Name: defaultSubNameAndPkg,
 		},
 	}, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+	phase, err = r.reconcileManifest(inst)
 	return phase, err
 }
 
+func (r *Reconciler) reconcileManifest(inst *v1alpha1.Installation) (v1alpha1.StatusPhase, error) {
+	//add all completed products to the manifest
+	for _, stage := range inst.Status.Stages {
```

Comment:
this seems the wrong place for this code?

Reply:
yep, agreed, I'll refactor it